pkg/autodiscovery/configresolver.go

```diff
@@ -326,6 +327,16 @@ func getPid(tplVar []byte, svc listeners.Service) ([]byte, error) {
 	return []byte(strconv.Itoa(pid)), nil
 }
 
+// getHostname returns the hostname of the service, to be used
+// when the IP is unavailable or erroneous
+func getHostname(tplVar []byte, svc listeners.Service) ([]byte, error) {
+	name, err := svc.GetHostname()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get hostname for service %s, skipping config - %s", svc.GetID(), err)
```

Comment:
nit: the errors shouldn't be lowercase? [see](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)